examples/support/mod.rs

```diff
@@ -364,43 +366,41 @@ pub fn gui(ui: &mut conrod::UiCell, ids: &Ids, app: &mut DemoApp) {
 /// glutin+glium event loop that works efficiently with conrod.
 #[cfg(feature="glium")]
 pub struct EventLoop {
-    ui_needs_update: bool,
-    last_update: std::time::Instant,
+    pair: Arc<(Mutex<bool>, Condvar)>,
 }
 
 #[cfg(feature="glium")]
 impl EventLoop {
 
     pub fn new() -> Self {
         EventLoop {
-            last_update: std::time::Instant::now(),
-            ui_needs_update: false,
+            pair: Arc::new((Mutex::new(false), Condvar::new())),
         }
     }
 
     /// Produce an iterator yielding all available events.
-    pub fn next(&mut self, display: &glium::Display) -> Vec<glium::glutin::Event> {
-        // We don't want to loop any faster than 60 FPS, so wait until it has been at least 16ms
-        // since the last yield.
-        let last_update = self.last_update;
-        let sixteen_ms = std::time::Duration::from_millis(16);
-        let duration_since_last_update = std::time::Instant::now().duration_since(last_update);
-        if duration_since_last_update < sixteen_ms {
-            std::thread::sleep(sixteen_ms - duration_since_last_update);
-        }
-
-        // Collect all pending events.
+    pub fn next(&self, display: &glium::Display) -> Vec<glium::glutin::Event> {
         let mut events = Vec::new();
-        events.extend(display.poll_events());
 
-        // If there are no events and the `Ui` does not need updating, wait for the next event.
-        if events.is_empty() && !self.ui_needs_update {
-            events.extend(display.wait_events().next());
+        // FIXME: This will busy loop at 60FPS, ideally there would be a way to have the glium
+        //        display also fire needs_update() calls and then we could bump the timeout
+        //        to something much higher or even just use wait()
```

Comment:
I think we might only get benefits from `Condvar` if the conrod loop is run on a separate thread from the `winit` events loop, which is certainly possible though a little more involved and probably a bit complicated for the examples (which this simple event loop implementation is for). I don't think there's any way for `winit` events to `notify` its own thread as they are waited for on the same thread that's being `wait`ed by the `Condvar`.